index.js

```diff
@@ -1,257 +1,229 @@
-    mime         = require('mime'),
-    fs           = require('fs');
-
-  class ClientDeploy extends S.classes.Plugin {
-
-    constructor() {
-      super();
-      this.name = 'serverless-client-s3';
-    }
-
-    registerActions() {
-      S.addAction(this.clientDeploy.bind(this), {
-        handler:       'clientDeploy',
-        description:   `Deploy your Serverless clients to S3 Website Bucket.`,
-        context:       'client',
-        contextAction: 'deploy',
-        options:       [
-          {
-            option:      'stage',
-            shortcut:    's',
-            description: 'stage to populate any variables'
-          }, {
-            option:      'region',
-            shortcut:    'r',
-            description: 'region to populate any variables'
+const path     = require('path');
+const BbPromise    = require('bluebird');
+const async        = require('async');
+const _            = require('lodash');
+const mime         = require('mime');
+const fs           = require('fs');
+
+class Client {
+  constructor(serverless, options){
+    this.serverless = serverless;
+    this.stage = options.stage || _.get(serverless, 'service.provider.stage')
+    this.region = options.region || _.get(serverless, 'service.provider.region');
+    this.provider = 'aws';
+    this.aws = this.serverless.getProvider(this.provider);
+
+    this.commands = {
+      client: {
+        usage: 'Generate and deploy clients',
+        lifecycleEvents:[
+          'client',
+          'deploy'
+        ],
+        commands: {
+          deploy: {
+            usage: 'Deploy serverless client code',
+            lifecycleEvents:[
+              'deploy'
+            ]
           }
-        ]
-      });
-      return BbPromise.resolve();
-    }
-
-    clientDeploy(evt) {
+        }
+      }
+    };
 
-      let _this     = this;
-      _this.evt     = evt;
 
-      // Flow
-      return _this._prompt()
-        .bind(_this)
-        .then(_this._validateAndPrepare)
-        .then(_this._processDeployment)
-        .then(function() {
+    this.hooks = {
+      'client:client': () => {
+        this.serverless.cli.log(this.commands.client.usage);
+      },
 
-          _this._spinner.stop(true);
-          SCli.log(`Finishing deployment...`);
+      'client:deploy:deploy': () => {
+        this._validateAndPrepare()
+          .then(this._processDeployment.bind(this));
+      }
+    };
+  }
 
-          // display friendly message after all async operations (file uploads) are finished
-          process.on('exit', function (){
-            SCli.log(`Successfully deployed client to: ${_this.bucketName}.s3-website-${_this.evt.options.region}.amazonaws.com`);
-          });
+  _validateAndPrepare() {
+    const Utils = this.serverless.utils;
+    const Error = this.serverless.classes.Error;
 
-          return _this.evt;
+    const _dist = 
+      (this.serverless.service.custom && this.serverless.service.custom.client) ? 
```

Comment:
This breaks the default case of not defining the distributionFolder since custom client will exist in order to define the bucketName. This should be changed to the following:

`const _dist =_.get(this.serverless, 'service.custom.client.distributionFolder', 'dist');`